kvstore/simplemap/simplemap.go

```diff
@@ -0,0 +1,148 @@
+package simplemap
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"slices"
+
+	"github.com/pokt-network/smt/kvstore"
+)
+
+var _ kvstore.KVStore = &SimpleMap{}
+
+const maxKeySize = 65000
+
+// InvalidKeyError is thrown when a key that does not exist is being accessed.
+type InvalidKeyError struct {
+	Key []byte
+}
+
+func (e *InvalidKeyError) Error() string {
```

Comment:
Can we add some godoc comments similar to the badger KVStore ones (copy paste will probably be enough with minor changes) to all the functions and structs in here.